extensions/ql-vscode/src/view/model-editor/ModelTypeTextbox.tsx

```diff
@@ -0,0 +1,63 @@
+import type { ChangeEvent } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import type {
+  ModeledMethod,
+  TypeModeledMethod,
+} from "../../model-editor/modeled-method";
+import { VSCodeTextField } from "@vscode/webview-ui-toolkit/react";
+import { useDebounceCallback } from "../common/useDebounceCallback";
+
+type Props = {
+  modeledMethod: TypeModeledMethod;
+  typeInfo: "path" | "relatedTypeName";
+  onChange: (modeledMethod: ModeledMethod) => void;
+};
+
+export const ModelTypeTextbox = ({
+  modeledMethod,
+  typeInfo,
+  onChange,
+}: Props): JSX.Element => {
+  const enabled = useMemo(() => modeledMethod, [modeledMethod]);
+  const [value, setValue] = useState<string | undefined>(
+    modeledMethod[typeInfo],
+  );
+
+  useEffect(() => {
+    if (modeledMethod) {
+      setValue(modeledMethod[typeInfo]);
+    }
```

Comment:
The `modeledMethod` shouldn't be undefined anymore:

```suggestion
    setValue(modeledMethod[typeInfo]);
```